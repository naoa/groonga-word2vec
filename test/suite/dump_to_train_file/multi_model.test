plugin_register word2vec/word2vec

table_create Tags TABLE_PAT_KEY ShortText

table_create Entries TABLE_NO_KEY
column_create Entries title COLUMN_SCALAR ShortText
column_create Entries tag COLUMN_SCALAR Tags
column_create Entries tags COLUMN_VECTOR Tags

load --table Entries
[
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "Database", "tag": "Server", "tags": ["MySQL", "PostgreSQL"]}
]

dump_to_train_file Entries title,tags --train_file "db/db_w2v.txt"
dump_to_train_file Entries title,tag --train_file "db/db2_w2v.txt"
word2vec_train --min_count 1 --train_file "db/db_w2v.txt" --output_file "db/db_w2v.bin"
word2vec_train --min_count 1 --train_file "db/db2_w2v.txt" --output_file "db/db2_w2v.bin"
word2vec_distance "Groonga" --file_path "db/db_w2v.bin"
word2vec_distance "Library" --file_path "db/db2_w2v.bin"
word2vec_distance "Groonga" --file_path "db/db_w2v.bin"
word2vec_distance "Library" --file_path "db/db2_w2v.bin"
