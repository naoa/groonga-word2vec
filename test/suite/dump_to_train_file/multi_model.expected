plugin_register word2vec/word2vec
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Entries tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
column_create Entries tags COLUMN_VECTOR Tags
[[0,0.0,0.0],true]
load --table Entries
[
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "FulltextSearch", "tag": "Library", "tags": ["Groonga", "Rroonga"]},
{"title": "Database", "tag": "Server", "tags": ["MySQL", "PostgreSQL"]}
]
[[0,0.0,0.0],6]
dump_to_train_file Entries title,tags --train_file "db/db_w2v.txt"
[[0,0.0,0.0],true]
dump_to_train_file Entries title,tag --train_file "db/db2_w2v.txt"
[[0,0.0,0.0],true]
word2vec_train --min_count 1 --train_file "db/db_w2v.txt" --output_file "db/db_w2v.bin"
[[0,0.0,0.0],true]
word2vec_train --min_count 1 --train_file "db/db2_w2v.txt" --output_file "db/db2_w2v.bin"
[[0,0.0,0.0],true]
word2vec_distance "Groonga" --file_path "db/db_w2v.bin"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      6
    ],
    [
      [
        "_key",
        "ShortText"
      ],
      [
        "_value",
        "Float"
      ]
    ],
    [
      "fulltextsearch",
      0.183403044939041
    ],
    [
      "</s>",
      0.0981750339269638
    ],
    [
      "mysql",
      -0.038426760584116
    ],
    [
      "rroonga",
      -0.041764497756958
    ],
    [
      "postgresql",
      -0.0477972477674484
    ],
    [
      "database",
      -0.128371581435204
    ]
  ]
]
word2vec_distance "Library" --file_path "db/db2_w2v.bin"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      4
    ],
    [
      [
        "_key",
        "ShortText"
      ],
      [
        "_value",
        "Float"
      ]
    ],
    [
      "fulltextsearch",
      0.183403044939041
    ],
    [
      "</s>",
      0.0981750339269638
    ],
    [
      "database",
      -0.041764497756958
    ],
    [
      "server",
      -0.128371581435204
    ]
  ]
]
word2vec_distance "Groonga" --file_path "db/db_w2v.bin"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      6
    ],
    [
      [
        "_key",
        "ShortText"
      ],
      [
        "_value",
        "Float"
      ]
    ],
    [
      "fulltextsearch",
      0.183403044939041
    ],
    [
      "</s>",
      0.0981750339269638
    ],
    [
      "mysql",
      -0.038426760584116
    ],
    [
      "rroonga",
      -0.041764497756958
    ],
    [
      "postgresql",
      -0.0477972477674484
    ],
    [
      "database",
      -0.128371581435204
    ]
  ]
]
word2vec_distance "Library" --file_path "db/db2_w2v.bin"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      4
    ],
    [
      [
        "_key",
        "ShortText"
      ],
      [
        "_value",
        "Float"
      ]
    ],
    [
      "fulltextsearch",
      0.183403044939041
    ],
    [
      "</s>",
      0.0981750339269638
    ],
    [
      "database",
      -0.041764497756958
    ],
    [
      "server",
      -0.128371581435204
    ]
  ]
]
